### DISCUSSION TITLE
"Folders" for the Vocabulary Builder

### DISCUSSION THREAD

**Discussion by eliasiturri** [koreader/koreader#12208] [Ideas]
I think it would be nice to be able to add words to different "folders" in the vocabulary builder. I read my books in several languages and having all the words stored under the same category makes the vocabulary builder not very useful when it's time to review them. This is an issue another friend of mine that uses koreader is having, so I expect it to affect more people.

I have done some changes to the code in the past for my personal use, so I wouldn't mind trying to put together a PR if this turns out to be a desired feature.

To avoid affecting users that don't need this, the "Add to vocabulary builder" button could be turned into two:

- The first one would let you add a word, but would request you to pick a "folder" or create a new one first.
- Once there is a "last folder" saved, the second button would allow you to quickly add the word to the last "folder" that was used.

Saving the words could be done like it is being done until now, adding a "folder name" field to the table and allowing the user to filter by folder name in the same way they are already filtering by book title.

I'm open to any suggestion or general directions on how to code it while affecting the whole project as little as possible.

P.S.: I'm not yet sure how to show a custom menu entry on the dictionary lookup, but it seems I can do it all just by adding a new plugin, so I will just clone the plugin, edit it and publish it as a new repository when I'm done. Any input is still appreciated.

**Comment by jonnyl2** [comment]
FWIW, the idea of having folders sounds good to me. Studying vocabulary by book is often not very practical, as one book can have a variety of different words that have little to do with one another. It would be nice to be able to categorize them by importance, situations, formality, language, etc., instead.
